File: Algorithms/Sorting_Algorithms/Bubble_Sort/BubbleSort.js
function swap(arr, first_Index, second_Index) {
  var temp = arr[first_Index];
  arr[first_Index] = arr[second_Index];
  arr[second_Index] = temp;
}

function bubbleSort(arr) {
  if (arr.length < 1) return 0;
  var arrayLength = arr.length;
  for (let i = 0; i < arrayLength; i++) {
    for (let j = 0, end = arrayLength - i; j < end; j++) {
      if (arr[j] > arr[j + 1]) {
        swap(arr, j, j + 1);
      }
    }
  }
  return arr;
}
console.log(bubbleSort([3, 0, 2, 5, -1, 4, 1]));

File: Algorithms/Sorting/Bubble_Sort/bubbleSort.cpp
#include<bits/stdc++.h>
using namespace std;

void bubble_sort(int arr[],int n)
{
    for(int i = 0;i<n;i++)
    {
        for(int j =0;j<n;j++)
        {
            if(arr[i] < arr[j])
            {
                swap(arr[i],arr[j]);
            }
        }
    }
}
int main()
{
    int n;
    cin>>n;
    int arr[n];
    for(int i =0;i<n;i++)cin>>arr[i];
    bubble_sort(arr,n);
    

}